### PR TITLE
Use single-port Dropwizard servers for all web resources

### PR DESCRIPTION
**Pull request by olim7t**
The idea is to centralize all admin features in health-checker: we already do it for metrics, and we should do the same for health checks.

Proof that the `--bind-to-listen-address` option still works as expected:
```
$ lsof -i TCP:8085

# without option:
NODE NAME
 TCP *:8085 (LISTEN)

# with option:
NODE NAME
 TCP 127.0.0.2:8085 (LISTEN)
```

### CODE REVIEW

**Review by dimas-b — APPROVED** [review]

**Comment by dimas-b** [comment]
Just wondering: why do we need admin connectors at all?

**Comment by olim7t** [comment]
> Just wondering: why do we need admin connectors at all?

Good point. We could expose metrics and health checks as regular resources in health-checker.

File: health-checker/src/main/java/io/stargate/health/Server.java [hunk 1]
```diff
@@ -70,6 +75,21 @@ protected void configure() {
...
+      DefaultServerFactory server =
+          (DefaultServerFactory) applicationConfiguration.getServerFactory();
+      ConnectorFactory connector = server.getAdminConnectors().get(0);
+      if (connector instanceof HttpConnectorFactory) {
+        ((HttpConnectorFactory) connector)
+            .setBindHost(System.getProperty("stargate.listen_address"));
+      } else {
+        throw new AssertionError("Unexpected connector class: " + connector.getClass().getName());
+      }
+    }
```
Comment (tomekl007) [inline health-checker/src/main/java/io/stargate/health/Server.java:90]: maybe add `log.debug` in the else branch to not "fail" silently?

**Review by dougwettlaufer — APPROVED** [review]

File: health-checker/src/main/java/io/stargate/health/Server.java (hunk 1 above)
Comment (olim7t) [reply health-checker/src/main/java/io/stargate/health/Server.java:90]: The exception should bubble up and appear in the Stargate logs I think.

**Comment by olim7t** [comment]
Following Dmitri's suggestion, I switched all modules to the "simple" Dropwizard server type, which uses only one port. The health-checker module exposes admin resources (metrics, health checks...) on that port. The other modules expose their application resources unchanged.

**Review by dimas-b — APPROVED** [review]

**Review by dimas-b — APPROVED** [review]

File: auth-api/src/main/java/io/stargate/auth/api/impl/AuthApiServerFactory.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.thread.ThreadPool;
+
+/**
+ * @see "META-INF/services/io.dropwizard.server.ServerFactory"
+ * @see "config.yaml"
+ */
+@JsonTypeName("auth-api")
+public class AuthApiServerFactory extends SimpleServerFactory {
```
Comment (olim7t) [inline auth-api/src/main/java/io/stargate/auth/api/impl/AuthApiServerFactory.java:34]: The admin resources were still exposed under `/admin` (default path), this completely removes them.

It's the exact same class for all 3 modules. I would have liked to factor it in `core`, but this introduces more issues with OSGi 😞

Comment (dimas-b) [reply auth-api/src/main/java/io/stargate/auth/api/impl/AuthApiServerFactory.java:34]: What kind of issues? :)

We could create a new module (e.g. `web-common`) and produce a _plain_ jar from it that would then be _embedded_ into the various OSGi bundles (unlike `core`, which is a bundle in itself). I believe this way would give us ability to reuse code without introducing dependency problems.

Comment (olim7t) [reply auth-api/src/main/java/io/stargate/auth/api/impl/AuthApiServerFactory.java:34]: I've created a follow-up ticket: #439.

**Pull request merged**